amazon-kinesis-client/src/main/java/software/amazon/kinesis/lifecycle/ShardConsumer.java

```diff
@@ -59,7 +59,13 @@ public class ShardConsumer {
     private final ShardConsumerArgument shardConsumerArgument;
     @NonNull
     private final Optional<Long> logWarningForTaskAfterMillis;
+
+    /**
+     * @deprecated unused; to be removed in a "major" version bump
+     */
+    @Deprecated
```

Comment:
Should we do this for **bufferSize** and **recordsPublisher** as well? Both those properties aren't in use either.